hack/ensure-golangci-lint.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+set -euo pipefail
+
+VERSION=1.41.1
+GOOS=$(go env GOOS)
+TARNAME=golangci-lint-$VERSION-$GOOS-amd64.tar.gz
+URL=https://github.com/golangci/golangci-lint/releases/download/v$VERSION/$TARNAME
+GOLANGCI_LINT=bin/golangci-lint
+
+case $GOOS in
```

Comment:
are you sure the checksums are correct? I didn't find any of those two hashes in the https://github.com/golangci/golangci-lint/releases/download/v1.41.1/golangci-lint-1.41.1-checksums.txt

Reply:
Those checksums are for the tarballs. But I'm verifying the checkum of the binary to simplify redownloading the binary when we update the version.